src/features/todo/todoSlice.jsx

```diff
@@ -22,9 +34,28 @@ export const productSlice = createSlice({
         deleteProduct: (state, action) => {
             state.products = state.products.filter((product) => product.id !== action.payload)
             localStorage.setItem('products', JSON.stringify(state.products));
+        },
+        changeNumberProduct: (state, action) => {
+            const { id, number } = action.payload;
+            const product = state.products.find((product) => product.id === id);
+            if (product) {
+                product.number = number;
+            }
+            localStorage.setItem('products', JSON.stringify(state.products));
         }
+
     }
 })
 
-export const { addProduct, deleteProduct } = productSlice.actions
+export const useTotalCost = (products) => {
+    const [totalCost, setTotalCost] = useState(calculateTotalCost(products));
+
+    useEffect(() => {
+        setTotalCost(calculateTotalCost(products));
+    }, [products]);
```

Comment:
no need to save `totalCost` in state. read carefully about [useSelector](https://redux.js.org/tutorials/fundamentals/part-5-ui-react#reading-state-from-the-store-with-useselector) until you fully understand it